client/engine/engine.go

```diff
@@ -424,28 +433,26 @@ func (e *Engine) constructObjectiveFromMessage(id protocols.ObjectiveId, ss stat
 
 	switch {
 	case directfund.IsDirectFundObjective(id):
-		dfo, err := directfund.ConstructFromState(true, ss.State(), *e.store.GetAddress())
+		dfo, err := directfund.ConstructFromState(false, ss.State(), *e.store.GetAddress())
```

Comment:
Now that we're introducing approval/rejection could we just remove the `preApprove` flag from `ConstructFromState` ?

Reply:
Actually, I briefly went down this path, and the size of the change grew quite large.

I ran into a couple of issues. One is:
- `preapprove=false` is probably the right default value
- however, when creating an objective due to an API event, we need to set it to `true`
- setting it to `true` via the `Approve` function is awkward, because the return value has a different type. many places that use the objective are expecting a specific objective struct, but `Approve` returns a `protocols.Objective` interface